custom/succeed/expressions.py

```diff
@@ -0,0 +1,32 @@
+from casexml.apps.case.models import CommCareCase
+from corehq.apps.userreports.specs import TypeProperty
+from dimagi.ext.jsonobject import JsonObject
+
+
+class ReferencedIdExpressionSpec(JsonObject):
+    type = TypeProperty('succeed_referenced_id')
+
+    def __call__(self, item, context=None):
+        referenced_id = item['indices'][0]['referenced_id']
```

Comment:
might also want to have this one fail more gracefully if there are any `KeyError`s